samples/test-dal-with-nunit/DalWithNunitSample/DALTests/PizzaServiceTests.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Threading.Tasks;
+using DAL;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using NUnit.Framework;
+
+namespace DALTests
+{
+    public class PizzaServiceTests : TestBase
+    {
+        private IPizzaService _pizzaService;
+
+        [OneTimeSetUp]
+        public void Initialization()
+        {
+            _pizzaService = new PizzaService(DbContext);
+        }
+
+        [Test]
+        public async Task GetAll_WhenPizzasListExists_ShouldReturnAllPizzas()
+        {
+            // Arrange
+            DbContext.Pizzas.AddRange(new Pizza[]
+            {
+                new Pizza { Name = "Pepperoni" },
+                new Pizza { Name = "Spacca Napoli" },
+                new Pizza { Name = "Gino Sorbilo" }
+            });
+            await DbContext.SaveChangesAsync();
+
+            // Act
+            var pizzas = await _pizzaService.GetAll();
+
+            // Assert
+            pizzas.Should().BeEquivalentTo(
+                new Pizza[]
+            {
+                new Pizza { Name = "Pepperoni" },
+                new Pizza { Name = "Spacca Napoli" },
+                new Pizza { Name = "Gino Sorbilo" }
+            }, options => options.Excluding(x => x.Id));
+        }
+
+        [Test]
+        public async Task GetById_WhenPizzaWithSpecifiedIdExists_ShouldReturnFoundPizza()
+        {
+            // Arrange
+            var pepperoni = new Pizza { Name = "Pepperoni" };
+            DbContext.Add(pepperoni);
+            await DbContext.SaveChangesAsync();
+
+            // Act
+            var result = await _pizzaService.GetById(pepperoni.Id);
+
+            // Assert
+            result.Should().BeEquivalentTo(new Pizza { Id = pepperoni.Id, Name = "Pepperoni" });
+        }
+
+        [Test]
+        public async Task Create_WhenPizzaSpecified_ShouldCreatePizza()
+        {
+            // Arrange & Act
+            await _pizzaService.Create(new Pizza { Name = "Pepperoni" });
+
+            // Assert
+            var result = await DbContext.Pizzas.FirstOrDefaultAsync(x => x.Name == "Pepperoni");
```

Comment:
This is unsafe since you're not gauranteeing you're querying from a different DbContext.  DbContext should be instantiated inside the test, not within the test class.

Reply:
Try to use IDisposable and instantiate new DbContext per test on its start with disposing of it on the test end.

Reply:
I'm using `TearDown` to clear the whole db, so I have clear data base state every test.
I don't think I should care about querying to different DbContexts

Reply:
No.  This is incorrect.  You're not querying the database, you're querying the context.  If your entity is tracked/your context supports tracking, the context will first look up the entity in the cache.  The only way to avoid EF not using the cache in this case is to override Equals and GetHashCode.

Again, your code is working by accident.

Reply:
Use EnsureTransaction and rollback per test to avoid ever actually committing to the db.  For most tests you don't need the transaction to actually commit, only conceptually commit in the sense it exercises the entire EF code stack.